src/RegistrationController.js

```diff
@@ -127,6 +127,21 @@ function (
         }
       });
     },
+    // @param url (Eg: ?fromURI=%2Fapp%2FUserHome&query=blah)
+    // returns {fromURI:'%2Fapp%2FUserHome', query: blah}
+    getJsonFromUrl: function(url) {
+      var query = url.substr(1);
```

Comment:
what happen if substr is empty?

Reply:
From the spec
https://www.w3schools.com/jsref/jsref_substr.asp
```If start is negative or larger than the length of the string, start is set to 0```
So empty string will be fine

Reply:
yep, when query is empty it will return empty object.